### PR TITLE
ENH: Add option to skip brain extraction

### LINKED ISSUE #12: Gracefully handle pre-skull-stripped T1w images
A number of subjects in ds000223 (e.g. [sub-07](https://s3.amazonaws.com/openneuro/ds000223/ds000223_R1.0.0/uncompressed/derivatives/mriqc/sub-07_T1w.html)) have pre-skull-stripped T1w images. ANTs skull-stripping on these datasets clips most superficial cortex, resulting in very poor preprocessing.

I propose adding a `--pre-skull-stripped` flag (or similar) to indicate that the anatomical workflow should assume the incoming image is already skull-stripped.

Alternately, we could do a test for a given percentage of zero voxels as evidence for skull-stripping, though this would need to be tested against defacers, as well.

### PR DESCRIPTION
**Pull request by mgxd**
Supercedes #116 
Fixes #12.

### CODE REVIEW

**Comment by mgxd** [comment]
cc @adelavega - I added a few documentation misses, and hopefully avoided missing any merges

File: smriprep/workflows/anatomical.py [hunk 1]
```diff
@@ -68,10 +72,11 @@ def init_anat_preproc_wf(
                 freesurfer=True,
                 hires=True,
                 longitudinal=False,
-                num_t1w=1,
+                t1w=['T1w.nii.gz'],
```
Comment (effigies) [inline smriprep/workflows/anatomical.py]: Maybe add to the run-up to `init_anat_preproc_wf`:

```Python
import nibabel as nb
import numpy as np
nb.Nifti1Image([[[1]]], np.eye(4)).to_filename('T1w.nii.gz')
```

Not sure if you can do any fancy fixture to make sure it's in a temporary directory. Or you can just remove it after.

**Comment by adelavega** [comment]
Looks reasonable to me, thanks!

File: smriprep/workflows/anatomical.py (hunk 1 above)
Comment (mgxd) [reply smriprep/workflows/anatomical.py]: we might be able to dust off the `wf_pre_code` option https://nipype.readthedocs.io/en/0.13.1/users/sphinx_ext.html#configuration-options, though i'm not entirely certain how to use it / we'll still need to remove the file

File: smriprep/cli/run.py [hunk 2]
```diff
@@ -87,6 +87,10 @@ def get_parser():
     g_ants.add_argument('--skull-strip-fixed-seed', action='store_true',
                         help='do not use a random seed for skull-stripping - will ensure '
                              'run-to-run replicability when used with --omp-nthreads 1')
+    g_ants.add_argument(
+        '--skip-brain-extraction', required=False, action='store',
+        default='auto', help='Skip the brain extraction workflow. '
```
Comment (mgxd) [inline smriprep/cli/run.py]: the help message should be expanded to reflect on each action's behavior

File: smriprep/workflows/anatomical.py [hunk 3]
```diff
@@ -104,6 +111,10 @@ def init_anat_preproc_wf(
         Enable debugging outputs
     name : :obj:`str`, optional
         Workflow name (default: anat_preproc_wf)
+    skip_brain_extraction : :obj:`str`
+        Skip ants brain extraction workflow, and instead use N4-only workflow
```
Comment (mgxd) [inline smriprep/workflows/anatomical.py]: docstring should also reflect choices

**Comment by mgxd** [comment]
I'm leaning towards including this in 20.1, but if there are strong sentiments we can cut smriprep 0.6 and include this in 20.0

**Comment by adelavega** [comment]
Awesome, thanks. No rush from me at all.

**Comment by effigies** [comment]
Okay. Let's not rush it then. I think we'll have a 0.5.2 release for fMRIPrep 20.0, and we can merge this afterwards.

File: smriprep/workflows/anatomical.py [hunk 4]
```diff
@@ -222,12 +233,34 @@ def init_anat_preproc_wf(
...
+        extreme planes for all (near) zeros.
+        """
+        def _check_img(img):
+            data = np.abs(nb.load(img).get_fdata(dtype=np.float32))
+            sidevals = data[0, :, :].sum() + data[-1, :, :].sum() + \
+                data[:, 0, :].sum() + data[:, -1, :].sum() + \
+                data[:, :, 0].sum() + data[:, :, -1].sum()
+            return sidevals < 10
+
+        return all(_check_img(img) for img in imgs)
```
Comment (oesteban) [inline smriprep/workflows/anatomical.py]: as we are delaying this definition, let's delay it to the latest possible (added suggestion below)
```suggestion
```

File: smriprep/workflows/anatomical.py [hunk 5]
```diff
@@ -222,12 +233,34 @@ def init_anat_preproc_wf(
...
+
+    if skull_strip_mode == 'auto':
+        skull_strip_mode = _is_skull_stripped(t1w)
```
Comment (oesteban) [inline smriprep/workflows/anatomical.py]:
```suggestion
        import numpy as np
        import nibabel as nb
        def _is_skull_stripped(imgs):
            """
            Check if T1w images are skull-stripped.
            
            An input is called skull-stripped if (almost) all voxels in extreme planes are zero.
            
            """
            def _check_img(img):
                data = np.abs(nb.load(img).get_fdata(dtype=np.float32))
                sidevals = data[0, :, :].sum() + data[-1, :, :].sum() + \
                    data[:, 0, :].sum() + data[:, -1, :].sum() + \
                    data[:, :, 0].sum() + data[:, :, -1].sum()
                return sidevals < 10

            return all(_check_img(img) for img in imgs)
        skull_strip_mode = _is_skull_stripped(t1w)
```

File: smriprep/workflows/anatomical.py [hunk 6]
```diff
@@ -3,6 +3,9 @@
 """Anatomical reference preprocessing workflows."""
 from pkg_resources import resource_filename as pkgr
 
+import numpy as np
+import nibabel as nb
+
```
Comment (oesteban) [inline smriprep/workflows/anatomical.py]: Please delay import
```suggestion
```

**Pull request merged**